ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ProfileClustersPage.tsx

```diff
@@ -1,44 +1,56 @@
+import { useParams } from 'react-router-dom';
+import { Divider, PageSection, Title } from '@patternfly/react-core';
 
-import { complianceEnhancedCoveragePath } from 'routePaths';
+import PageTitle from 'Components/PageTitle';
+import useRestQuery from 'hooks/useRestQuery';
+import { getComplianceClusterStats } from 'services/ComplianceResultsStatsService';
+import { getTableUIState } from 'utils/getTableUIState';
 
-import CoveragesToggleGroup from './CoveragesToggleGroup';
 import CoveragesPageHeader from './CoveragesPageHeader';
+import CoveragesToggleGroup from './CoveragesToggleGroup';
+import ProfileClustersTable from './ProfileClustersTable';
 
 function ProfileClustersPage() {
-    const history = useHistory();
     const { profileName } = useParams();
+    const [currentDatetime, setCurrentDatetime] = useState<Date>(new Date());
+
+    const fetchProfileClusters = useCallback(
+        () => getComplianceClusterStats(profileName),
+        [profileName]
+    );
+    const { data: profileClusters, loading: isLoading, error } = useRestQuery(fetchProfileClusters);
+
+    const tableState = getTableUIState({
+        isLoading,
+        data: profileClusters?.scanStats,
+        error,
+        searchFilter: {},
+    });
+
+    useEffect(() => {
+        if (profileClusters) {
+            setCurrentDatetime(new Date());
+        }
+    }, [profileClusters]);
 
     return (
         <>
+            <PageTitle title="Compliance coverage - Profile clusters" />
             <CoveragesPageHeader />
             <PageSection>
                 <CoveragesToggleGroup tableView="clusters" />
             </PageSection>
-            <PageSection variant="light">
-                <div>ProfileClustersPage</div>
-                <Button
-                    onClick={() => {
-                        history.push(
-                            `${complianceEnhancedCoveragePath}/profiles/${profileName}/checks`
-                        );
-                    }}
-                    variant="primary"
-                >
-                    Go to all checks (ProfileChecksPage)
-                </Button>
-                <Button
-                    onClick={() => {
-                        history.push(
-                            `${complianceEnhancedCoveragePath}/profiles/${profileName}/clusters/test`
-                        );
-                    }}
-                    variant="primary"
-                >
-                    Go to single cluster (ClusterDetailsPage)
-                </Button>
+            <PageSection variant="default">
+                <PageSection variant="light" component="div">
+                    <Title headingLevel="h2">Profile results</Title>
+                    <Divider />
+                    <ProfileClustersTable
+                        currentDatetime={currentDatetime}
+                        profileName={profileName}
+                        tableState={tableState}
```

Comment:
What do you think about keeping the content of `ProfileClustersTable` within this component? Sometimes I've found that creating too many layers to components in the beginning can make it hard to follow and debug. You need to jump around a bit to find what you need. For this case, I think it would make sense to keep things centralized for now. 

Example 1:
```
<Page>
    <PageHeader />
    <Table />
</Page>
```



Example 2:
```
<Page>
    <PageHeader />
    <Table>
        <TableHead>
            <Tr>
                <Th>...</Th>
                <Th>...</Th>
                <Th>...</Th>
            </Tr>
        </TableHead>
        <TableBody>
           <Tr>
                <Td>...</Td>
                <Td>...</Td>
                <Td>...</Td>
            </Tr>
        </TableBody>
    </Table>
</Page>
```

For the first case, you'd have to dig into that component to see what's going on. In the second case, you can see everything at a first glance.

Reply:
It's a good thought and I agree that creating too many layers can make things complex but I think I'd argue the opposite here. This feels like a clear separation of concerns and a designated "Table" component would hopefully increase readability. Otherwise, all table-specific information would clutter up the "Page" component.

Reply:
Hmm, can you explain why you think it separates concerns? It seems to only render the table without any state logic. Because of that, I believe it would be more readable in the parent component, as this keeps the logic inline and avoids context switching to understand the component's purpose. It feels like an unnecessary abstraction right now. But I'm sure you have a bigger picture in mind for the component, so I'm good with keeping it like this. Just wanted to raise my concerns.